### ISSUE TITLE
EssentialsX Chat sees Primary groups!

### ISSUE THREAD

**Issue by Ironic8b498** [EssentialsX/Essentials#1132]
I would like to know if I can set it so EssentialsX Chat can see all groups, not just primary group. I run a bungeecord network with 6 server which have separate ranking system. What ever server the player ranks up on last that group is the primary, so all the other servers chat format is broken for them

**Comment by mdcfe** [comment]
EssentialsX doesn't do Bungee-wide chat formatting.

**Comment by Ironic8b498** [comment]
I know, don't need bungee-wide,  I use Bungee Essentials for that. I just need EssentialsX  to see the right group the player is in for that server.  On the survival server the ranks are 0, 1, 2, 3, 4 and VIP if the player just ranked up to rank 2, and they switch to Vanilla which uses, citizen, traveler, farmer, miner, hunter and Pro since I use group-format citizen: '&f[&7{GROUP}&f] <&7{DISPLAYNAME}>&8:&f {MESSAGE}' EssentialsX don't see the player in a group since the primary group they are in is not in list. If I add the group they are in it works great. Had same issue with autorank which had a setting so it would check all groups a player was in not just primary.

**Comment by mdcfe** [comment]
I think only using the primary group is intended; otherwise, how could EssX decide which non-primary group is the one to use for formatting?

**Comment by Ironic8b498** [comment]
the same way it filters to see if the group is in the list of group-format. if group is not in list then they get default format. just would have to check each group the player is in. Maybe this will get fixed with PEX 2.0 when I figure out how to get it to work. Was just looking for a fix right now.

**Comment by mdcfe** [comment]
If this is the use case, wouldn't group formatting suffice?

**Comment by Ironic8b498** [comment]
no cause the ranks from one server have no perms on other servers, so why display their ranks in chat

**Comment by mdcfe** [comment]
I don't understand what the issue is, how your setup works or how this is EssX related. Are your permissions synced between all servers? What plugin is responsible for Bungee-wide chat?

To answer your initial question: no, I don't think what you were asking can be changed from default behaviour.

**Comment by Ironic8b498** [comment]
Well I was not able to figure out PEX 2.0, but I made changes to the PEX 1.23.4 I'm running, to allow the Autorank plugin to work, but I still have broken chat format. If I use SQL on PEX the chat format breaks since players are in more then one group. If a player chooses to play on all my servers, they will belong to 7 groups. so if I ran "/pex user <playername> group list" I will see a list of 7 groups. if a player is on a server the server ignores all but one group for perms and Autorank, etc.. For Autorank I was able to change a setting in the config which allow the plugin to read all the groups not just primary group.  That way if the primary group was not in the ranking path.yml it will match one of the others in group list. I have tested all my servers by setting a player to the group I want to test in chat format and it works fine. If the player has more then one group, the group that's set to primary in vault, if that group is in group-formats it works on the server. Hope this helps up understand what I'm trying to do
